server/src/test/java/org/elasticsearch/transport/RemoteClusterSettingsTests.java

```diff
@@ -71,9 +76,48 @@ public void testSeedsDefault() {
         assertThat(REMOTE_CLUSTER_SEEDS.getConcreteSettingForNamespace(alias).get(Settings.EMPTY), emptyCollectionOf(String.class));
     }
 
+    public void testCredentialsDefault() {
+        assumeTrue("Skipped test because CCx2 feature flag is not enabled", TcpTransport.isUntrustedRemoteClusterEnabled());
+        final String alias = randomAlphaOfLength(8);
+        assertThat(REMOTE_CLUSTER_AUTHORIZATION.getConcreteSettingForNamespace(alias).get(Settings.EMPTY), equalTo(""));
+    }
+
     public void testProxyDefault() {
         final String alias = randomAlphaOfLength(8);
         assertThat(REMOTE_CLUSTERS_PROXY.getConcreteSettingForNamespace(alias).get(Settings.EMPTY), equalTo(""));
     }
 
+    public void testRemoteClusterEmptyOrNullApiKey() {
+        assumeTrue("Skipped test because CCx2 feature flag is not enabled", TcpTransport.isUntrustedRemoteClusterEnabled());
+        // simple validation
+        Settings settings = Settings.builder()
+            .put("cluster.remote.cluster1.authorization", "apikey")
+            .put("cluster.remote.cluster2.authorization", "")
+            .put("cluster.remote.cluster3.authorization", (String) null)
+            .build();
+        try {
+            SniffConnectionStrategy.REMOTE_CLUSTER_AUTHORIZATION.getAllConcreteSettings(settings).forEach(setting -> setting.get(settings));
+        } catch (Throwable t) {
+            fail("Cluster Settings must be able to accept a null, empty, or non-empty string. Exception: " + t.getMessage());
+        }
+    }
+
+    public void testRemoteClusterAuthorizationSettingDependencies() {
+        assumeTrue("Skipped test because CCx2 feature flag is not enabled", TcpTransport.isUntrustedRemoteClusterEnabled());
+        AbstractScopedSettings service = new ClusterSettings(
+            Settings.EMPTY,
+            new HashSet<>(Arrays.asList(SniffConnectionStrategy.REMOTE_CLUSTER_SEEDS, SniffConnectionStrategy.REMOTE_CLUSTER_AUTHORIZATION))
+        );
+        {
+            Settings missingDependentSetting = Settings.builder().put("cluster.remote.foo.authorization", randomBoolean()).build();
+            IllegalArgumentException iae = expectThrows(
+                IllegalArgumentException.class,
+                () -> service.validate(missingDependentSetting, true)
+            );
+            assertEquals(
+                "missing required setting [cluster.remote.foo.seeds] for setting [cluster.remote.foo.authorization]",
```

Comment:
similar to above , i don't think seeds should be a dependent of authorization. We should have some validattion, but it needs to support both proxy and sniff.

Reply:
Removed the seeds dependency and the corresponding test. Resolved.